frontend/src/components/DevicesBrowser.vue

```diff
@@ -317,6 +323,17 @@ export default {
 
             return columns
         },
+        filteredDevices () {
+            let filteredDevices = []
+            if (!this.filter) {
+                filteredDevices = Array.from(this.devices.values())
+            } else {
+                filteredDevices = Array.from(this.devices.values()).filter((d) => {
+                    return this.filter.devices.includes(d.id)
```

Comment:
By converting the map to an array, then looping over the array searching `this.filter.devices` each time for each ID, this will end up `N^2`.

Given this.devices is already a `Map`, something like the below will be significantly faster:

```javascript
this.filter.devices.forEach((device) {
  filteredDevices.push(this.devices.get(device.id))
})
```